#! /bin/sh /usr/share/dpatch/dpatch-run
## 25_support_lvm_for_live-media.dpatch by  <mru@grml.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Support lvm and raid devices for live-media

@DPATCH@

--- a/scripts/live
+++ b/scripts/live
@@ -1669,6 +1669,29 @@
 			umount $mountpoint
 		fi
 	fi
+
+	IFS=","
+	for device in ${devname} ; do
+		case "$device" in
+			*mapper*)  # add lvm support
+			if [ -x /scripts/local-top/lvm2 ] ; then
+				ROOT="$device" resume="" /scripts/local-top/lvm2
+			fi
+			;;
+			/dev/md*)
+			if [ -x /scripts/local-top/mdadm ] ; then
+				cp /conf/conf.d/md /conf/conf.d/md.orig
+				echo "MD_DEVS=$device " >> /conf/conf.d/md
+				/scripts/local-top/mdadm
+				mv /conf/conf.d/md.orig /conf/conf.d/md
+			fi
+			;;
+		esac
+	done
+	unset IFS
+
+	[ -n "$device" ] && devname="$device"
+
 	[ -e "$devname" ] || continue
 
 	if [ -n "${LIVE_MEDIA_OFFSET}" ]
